py/nightwatch/qa/snr.py

```diff
@@ -115,82 +113,164 @@ def run(self, indir):
                     qframe = qframes[c]
                     iiband[c] = (qframe.wave[0][0] < self.rwave)
                     iiband[c] &= (self.rwave < qframe.wave[0][-1])
+                    
+            argslist = [(self, qframes, fmap, f, fiber, iiband, night, expid, spectro, stars, qsos) for f, fiber in enumerate(fmap["FIBER"])]
```

Comment:
A possible performance problem is that every process is getting the full `qframes` and `fmap` for all 500 spectra, and then using `f` to pick out just one spectrum of interest per process.  With multiprocessing, this memory is *copied*, not just referenced like normal python.  i.e. it could be thrashing memory or otherwise just copying a lot more data between the processes than is necessary.  I think the solution here would be to make a new version of `get_dico` that takes just the data for a single spectrum instead of the data for all spectra and an index `f`.

Reply:
Making these changes, the snr qa code is now running in a reasonable (and faster than before) amount of time- 20 seconds versus 60 seconds.